### PR TITLE
Access control vulnerability updated.

### PR DESCRIPTION
**Pull request by 0xSandyy**
## Related Issue

- Closes: https://github.com/kadenzipfel/smart-contract-vulnerabilities/issues/85

## Checklist

- [x] I have read and followed the [style guide](https://github.com/kadenzipfel/smart-contract-vulnerabilities/blob/master/style-guide.md)

### Describe the changes you've made:
I updated the insufficient access control vulnerability by adding some access control vulnerabilities and some mitigation strategies/best practices.

## Type of change

Select the appropriate checkbox:

- [ ] Bug fix (fixing an issue with existing vulnerability data)
- [ ] New feature (adding a new vulnerability or category)
- [x] Documentation update (improving existing information)

### CODE REVIEW

File: vulnerabilities/insufficient-access-control.md [hunk 1]
```diff
@@ -1,3 +1,159 @@
...
+The ``initContract()`` function sets the caller as the owner but lacks any checks to prevent re-initialization. An attacker can call this function to take ownership of the contract, thereby gaining control over its funds and functionalities.
+
+#### Overpowered Roles
+```solidity
+// Using OpenZeppelin's Ownable library
+function criticalFunction() public onlyOwner {
+    // Critical logic here
+}
+```
+If the contract assigns multiple roles with ``onlyOwner`` privileges, it increases the attack surface. An attacker compromising a single owner account could execute critical functions.
```
Comment (kadenzipfel) [inline vulnerabilities/insufficient-access-control.md:22]: This shouldn't be possible with OZ Ownable though

**Review by kadenzipfel — CHANGES_REQUESTED** [review]
Tbh I feel like most of this can really be simplified. The core of the issue mostly just comes down to lack of access control being the problem and adding access control being the solution. The additional explanations kinda take away from it and are mostly not relevant to the core issue

File: vulnerabilities/insufficient-access-control.md (hunk 1 above)
Comment (0xSandyy) [reply vulnerabilities/insufficient-access-control.md:22]: My bad. OZ Ownable is the fix for role based access control with multiple onlyOwner privileges. Will fix.

**Comment by 0xSandyy** [comment]
Yeah, I overdid it. How about adding a simple example like a ``withdraw()`` function or a critical function which has no access control with a fix using a require / onlyOwner modifier? Simple example to showcase lack of access control and it's fix.

File: vulnerabilities/insufficient-access-control.md [hunk 2]
```diff
@@ -1,3 +1,159 @@
...
+
+    modifier onlyWhitelistedMinter() {
+        require(whitelistedMinters[msg.sender], "Only whitelisted minters can call this function");
+        _;
+    }
+}
+```
+In the above contract, we’ve used ``whitelisting`` by maintaining a mapping of addresses that are authorized to mint NFTs. The ``mintNFT`` function now has the ``onlyWhitelistedMinter`` modifier, which makes sure that only addresses on the whitelist can call this function. Additionally, the contract ``owner`` can manage the whitelist using the ``addToWhitelist`` and ``removeFromWhitelist`` functions.
+
+### Sources
```
Comment (rakesh0x7) [inline vulnerabilities/insufficient-access-control.md:155]: @0xSandyy, Could you please convert links to hyper links

**Review by rakesh0x7 — CHANGES_REQUESTED** [review]

File: vulnerabilities/insufficient-access-control.md [hunk 3]
```diff
@@ -1,3 +1,159 @@
...
+- https://metaschool.so/articles/access-control-vulnerabilities-in-smart-contracts/
+- https://medium.com/rektify-ai/how-to-mitigate-access-control-vulnerability-6df74c82af98
+- https://github.com/Quillhash/Solidity-Attack-Vectors/blob/main/data/1.md
```
Comment (rakesh0x7) [inline vulnerabilities/insufficient-access-control.md:158]: @0xSandyy, Could you please convert links to hyper links

**Review by rakesh0x7 — CHANGES_REQUESTED** [review]

File: vulnerabilities/insufficient-access-control.md (hunk 3 above)
Comment (0xSandyy) [reply vulnerabilities/insufficient-access-control.md:158]: Thanks for the remainder!

**Comment by kadenzipfel** [comment]
Replaced by: https://github.com/kadenzipfel/smart-contract-vulnerabilities/pull/101

**Pull request closed without merging**